packages/protocol/script/DeployGraduation.s.sol

```diff
@@ -0,0 +1,117 @@
+import { GraduationHook } from "../contracts/uniswap/GraduationHook.sol";
+import { HelperConfig } from "./HelperConfig.s.sol";
+import { MoxieBondingCurveV3 } from "../contracts/MoxieBondingCurveV3.sol";
+import { ProxyAdmin } from "openzeppelin-contracts/contracts/proxy/transparent/ProxyAdmin.sol";
+import { IPositionManager } from "@uniswap/briefcase/src/protocols/v4-periphery/interfaces/IPositionManager.sol";
+import { IUniversalRouter } from "@uniswap/briefcase/src/protocols/universal-router/interfaces/IUniversalRouter.sol";
+import { ITransparentUpgradeableProxy } from "openzeppelin-contracts/contracts/proxy/transparent/TransparentUpgradeableProxy.sol";
+import { HookMiner } from "../contracts/tests/UniswapDeployer.sol";
+import { MoxieToken } from "../contracts/tokens/MoxieToken.sol";
+import { TokenManager } from "../contracts/TokenManager.sol";
+
+import { IERC20Extended } from "../contracts/interfaces/IERC20Extended.sol";
+
+contract DeployGraduation is HelperConfig {
+    function deployHook(
+        bytes memory creationCode,
+        bytes memory constructorArgs,
+        bytes32 salt
+    ) public returns (address hookAddress) {
+        bytes memory initCode = abi.encodePacked(creationCode, constructorArgs);
+        assembly ("memory-safe") {
+            let ptr := mload(0x40)
+            let success := create2(0, add(initCode, 0x20), mload(initCode), salt)
+
+            if iszero(success) {
+                // Get the size of the returned error message
+                let errorSize := returndatasize()
+                // Copy the error message to memory
+                returndatacopy(ptr, 0, errorSize)
+                // Revert with the error message
+                revert(ptr, errorSize)
+            }
+
+            hookAddress := success
+        }
+        return hookAddress;
+    }
+
+    function run() public {
+        (
+            uint256 deployerKey,
+            uint256 ownerKey,
+            ,
+            uint256 proxyAdminOwnerAccount
+        ) = deriveKeys();
+        address owner = vm.addr(ownerKey);
+
+        vm.startBroadcast(deployerKey);
+        bytes memory graduationConstructorArgs = abi.encode(
+            currentNetworkConfig.poolManager,
+            currentNetworkConfig.moxieBondingCurveInstance
+        );
+        (address graduationHookAddress, bytes32 salt) = HookMiner.find(
+            currentNetworkConfig.deterministicDeploymentProxy,
+            4096, // after initialize hook flag
+            type(GraduationHook).creationCode,
+            graduationConstructorArgs
+        );
+        console.log("Graduation hook address: %s", graduationHookAddress);
+        address graduationHook = deployHook(
+            type(GraduationHook).creationCode,
+            graduationConstructorArgs,
+            salt
+        );
+        console.log("Graduation hook deployed to %s", graduationHook);
+
+        MoxieBondingCurveV3 moxieBondingCurveV3 = new MoxieBondingCurveV3();
```

Comment:
can you rename it as moxieBondingCurveV3MasterCopy

Reply:
done